src/OSPSuite.Core/Domain/ModelCoreSimulationExtensions.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Domain.Builder;
+
+namespace OSPSuite.Core.Domain
+{
+   public static class ModelCoreSimulationExtensions
+   {
+      public static IReadOnlyList<ApplicationParameters> AllApplicationParametersOrderedByStartTimeFor(this IModelCoreSimulation simulation,
+         string moleculeName)
+      {
+         var endTime = simulation.EndTime ?? 0;
+         var allApplications = allApplicationsForMolecule(simulation, moleculeName);
+
+         return allApplicationParametersIn(allApplications, Constants.Parameters.START_TIME)
+            .Where(x => x.Value <= endTime)
+            .OrderBy(x => x.Value)
+            .Select(x => new ApplicationParameters(x))
+            .ToArray(); // Return array here to be compatible with R
+      }
+
+      private static IReadOnlyList<IContainer> allApplicationsForMolecule(IModelCoreSimulation simulation, string moleculeName)
+      {
+         var applicationEventGroup = simulation.Model.Root.GetChildren<IEventGroup>().ToList();
+         if (!applicationEventGroup.Any())
+            return new List<IContainer>();
+
+         var allApplications = applicationEventGroup.SelectMany(x => x.GetAllChildren<IContainer>(c => c.ContainerType == ContainerType.Application))
+            .ToList();
+
+         return getApplicationsForAppliedAncestorMolecule(simulation.Reactions, moleculeName, allApplications);
+      }
+
+      private static IReadOnlyList<IContainer> getApplicationsForAppliedAncestorMolecule(IEnumerable<IReactionBuilder> reactions, string moleculeName,
+         IReadOnlyList<IContainer> allApplications)
```

Comment:
ok, it's the same code as before and maybe I never looked at it... but I don't understand this function at all. 
* If my simulation has no reactions at all - it will return empty list and the calling function`allApplicationsForMolecule(IModelCoreSimulation simulation, string moleculeName)` will also return empty list - even if there are applications of `moleculeName`
* If I have a reaction A==>B then `allApplicationsForMolecule(simulation, "B") `will return Applications of A?

Reply:
@Yuri05 What are you talking about :)

The first call of the one and only public method is 

````
    var applicationsForAppliedAncestorMolecule =
            allApplications.Where(c => c.GetSingleChildByName<IMoleculeAmount>(moleculeName) != null).ToList();

         // If there are any applications of this molecule, use them
         if (applicationsForAppliedAncestorMolecule.Any())
            return applicationsForAppliedAncestorMolecule;
```

so if we found an application, we use it

Arguably, the test with reactions can be done later but a building block should NEVER be null. At most it should be empty

Reply:
If I have a reaction A==>B then allApplicationsForMolecule(simulation, "B") will return Applications of A?

NO only if there are no application producing B

So in short
* NO
* NO